vscode/src/test/analyzerResults.test.ts

```diff
@@ -1,27 +1,24 @@
 import * as assert from "assert";
-import * as vscode from "vscode";
+import { DiagnosticSeverity } from "vscode";
 import * as path from "path";
-import { processIncidents, readYamlFile } from "../client/analyzerResults";
+import { processIncidents, readYamlFile } from "../data/analyzerResults";
 import { RuleSet } from "@shared/types";
 
 suite("Extension Test Suite", () => {
   test("processIncidents should populate diagnostics correctly", () => {
     const filePath = path.resolve(__dirname, "./testData/output-data.yaml");
     const ruleSets: RuleSet[] | undefined = readYamlFile(filePath);
     assert.ok(ruleSets, "RuleSets should be loaded from YAML file");
-    const diagnosticsMap = new Map<string, vscode.Diagnostic[]>();
-    processIncidents(ruleSets!, diagnosticsMap);
+    const results = processIncidents(ruleSets!);
     assert.ok(
-      diagnosticsMap.has("file:///opt/input/source/src/main/webapp/WEB-INF/web.xml"),
+      results.every(([uri]) => uri.fsPath === "/opt/input/source/src/main/webapp/WEB-INF/web.xml"),
```

Comment:
Wonder if we need to normalize the path here to be platform agnostic? To satisfy CI we could try path.posix and path.normalize.

Reply:
good catch @ibolton336 ! It was the path problem. After I've refactored the test the assertion failure was clear:
```
   AssertionError [ERR_ASSERTION]: web.xml should have 4 diagnostics
      + expected - actual

       [
...
      -  "\\opt\\input\\source\\src\\main\\webapp\\WEB-INF\\web.xml"  
      +  "/opt/input/source/src/main/webapp/WEB-INF/web.xml"
...
       ]
```
`normalize()` did'nt work but split on `path.sep` did.